pkg/agent/flowexporter/connections/conntrack.go

```diff
@@ -59,7 +57,7 @@ func filterAntreaConns(conns []*flowexporter.Connection, nodeConfig *config.Node
 		// Therefore, we ignore the connection with ClusterIP and keep the connection with Endpoint Pod IP.
 		// Conntrack flows will be different for Pod-to-Service flows w/ Antrea-proxy. This implementation will be simpler, when the
 		// Antrea proxy is supported.
-		if serviceCIDR.Contains(srcIP) || serviceCIDR.Contains(dstIP) {
+		if !nodeConfig.PodCIDR.Contains(dstIP) {
 			klog.V(4).Infof("Detected a flow with Cluster IP :%v", conn)
 			continue
 		}
```

Comment:
not introduced by this PR, but above I see this code:

```go
		// Only get Pod-to-Pod flows.
		if srcIP.Equal(nodeConfig.GatewayConfig.IP) || dstIP.Equal(nodeConfig.GatewayConfig.IP) {
			klog.V(4).Infof("Detected flow through gateway :%v", conn)
			continue
		}
```

I'm wondering if the code and / or comment are incorrect. What about Pod-to-External traffic? The check will not filter them out, but they don't qualify as Pod-to-Pod traffic. It seems to me that we still want to export information about Pod-to-External flows, but I may be missing something.

Reply:
I did a small experiment with a nodePort service to expose a service to the external networks. 
I see that the related flows in Antrea zone ID and default zone ID (iptables for SNAT) go through the gateway of local podCIDR.

```
tcp      6 86362 ESTABLISHED src=192.168.77.1 dst=192.168.77.101 sport=55265 dport=30007 packets=6 bytes=1216 src=10.10.1.59 dst=10.10.1.1 sport=5601 dport=33564 packets=5 bytes=798 [ASSURED] mark=0 delta-time=92 use=1
tcp      6 86362 ESTABLISHED src=10.10.1.1 dst=10.10.1.59 sport=33564 dport=5601 packets=6 bytes=1216 src=10.10.1.59 dst=10.10.1.1 sport=5601 dport=33564 packets=5 bytes=798 [ASSURED] mark=32 zone=65520 delta-time=92 use=1
```

This is when an external IP tries to reach the nodePort service in K8s cluster. I presume this to be the Pod-To-External traffic. Let me know if I am wrong. 

As we are focusing only on flows in Antrea zoneID for now, these flows get filtered out. I agree that we have to support exporting information about Pod-To-External flows. Revised the existing comment and added a TODO for Pod-To-External traffic as this needs some more investigation and have to consider flows in default ZoneID.

Reply:
By Pod-to-External, I meant some connection initiated by the Pod, not a connection targeted at a NodePort Service, for which there will be SNAT & DNAT. In the example below, I am making an HTTP request to `www.google.com`.

```bash
tcp      6 116 TIME_WAIT src=10.10.1.65 dst=172.217.26.164 sport=43732 dport=80 packets=17 bytes=982 src=172.217.26.164 dst=10.0.2.15 sport=80 dport=43732 packets=13 bytes=15549 [ASSURED] mark=0 delta-time=5 use=1
tcp      6 116 TIME_WAIT src=10.10.1.65 dst=172.217.26.164 sport=43732 dport=80 packets=17 bytes=982 src=172.217.26.164 dst=10.10.1.65 sport=80 dport=43732 packets=18 bytes=15749 [ASSURED] mark=0 zone=65520 delta-time=5 use=1
```

Notice the second conntrack entry, which is committed in the Antrea CT zone. I think based on the filtering code you have, this connection - Pod (`10.10.1.65`) to External (`172.217.26.164`) - will not be discarded, as the gateway's IP address is not involved. Is that the intention or not? Unless I am missing something, either the code logic or the comment needs to be updated?

Reply:
Thanks for the clarification and the info on Pod-To-External flows. Yes, I will update the comment accordingly. 

I was thinking we should also add a new IPFIX field to determine the type of the flows: Pod-To-Pod, Pod-To-Service, Pod-To-External, and other types as well. IPFIX collectors can use this type to distinguish the flows and helps to render the dashboards too. If this sounds good, will keep a note to take care of this in separate PR.